src/F3DAudio.c

```diff
@@ -28,174 +28,1338 @@
+        if (a0 < a1)
+        {
+            if (emitterAzimuth >= a0 && emitterAzimuth < a1)
+            {
+                break;
+            }
+        }
+        else
+        {
+            if (emitterAzimuth >= a0 || emitterAzimuth < a1)
+            {
+                break;
+            }
+        }
+    }
+    SDL_assert(emitterAzimuth >= a0 || emitterAzimuth < a1);
+
+    if (skipCenter)
+    {
+        if (a0 == 0.0f)
+        {
+            if (i == 0)
+            {
+                i = config->numNonLFSpeakers - 1;
+            }
+            else
+            {
+                i -= 1;
+            }
+        }
+        else if (a1 == 0.0f)
+        {
+            nexti += 1;
+            if (nexti >= config->numNonLFSpeakers)
+            {
+                nexti -= config->numNonLFSpeakers;
+            }
+        }
+    }
+    speakerInfo[0] = &config->speakers[i];
+    speakerInfo[1] = &config->speakers[nexti];
+}
+
+/* ComputeInnerRadiusDiffusionFactors is a utility function that returns how energy dissipates to the speakers, given the radial distance between the emitter and the listener and the (optionally 0) InnerRadius distance. It returns 3 floats, via the diffusionFactors array, that say how much energy (after distance attenuation) will need to be distributed between each of the following cases:
+ * - SPEAKERS_ALL for all (non-LF) speakers, _INCLUDING_ the MATCHING and OPPOSITE.
+ * - SPEAKERS_OPPOSITE corresponds to the two speakers OPPOSITE the emitter.
+ * - SPEAKERS_MATCHING corresponds to the two speakers closest to the emitter.
+ *
+ * For a distance below a certain threshold (DISTANCE_EQUAL_ENERGY), all speakers receive equal energy.
+ * Above that, the amount that all speakers receive decreases linearly as radial distance increases, up until InnerRadius / 2. (If InnerRadius is null, we use MINIMUM_INNER_RADIUS.)
+ * At the same time, both opposite and matching speakers start to receive sound (in addition to the energy they receive from the aforementioned "all speakers" linear law) according to some (unknown as of known) quadratic law, that is currently emulated with a LERP. This is true up until InnerRadius.
+ * Above InnerRadius, only the two matching speakers receive sound.
+ */
+
+/* These constants were estimated roughly, by trial and error. */
+// TODO: determine them more accurately.
+#define DIFFUSION_DISTANCE_EQUAL_ENERGY 1e-7f
+#define DIFFUSION_DISTANCE_MINIMUM_INNER_RADIUS 4e-7f
+
+enum {
+    DIFFUSION_SPEAKERS_ALL = 0,
+    DIFFUSION_SPEAKERS_MATCHING = 1,
+    DIFFUSION_SPEAKERS_OPPOSITE = 2
+};
+
+/* Adrien: determined experimentally; this is the midpoint value, i.e. the value at 0.5 for the matching speakers, used for the standard diffusion curve. Note: it is SUSPICIOUSLY close to 1/sqrt(2), but I haven't figured out why. */
+#define DIFFUSION_LERP_MIDPOINT_VALUE 0.707107f
+
+static void ComputeInnerRadiusDiffusionFactors(float radialDistance, float InnerRadius, float *diffusionFactors)
```

Comment:
Would this function be the one where InnerRadius affects the speaker output? Just trying to narrow this down in case someone wants to step through this.

Reply:
Yes, that is the most probable one. I am still super busy until Sunday but I promise to look at it next week.

Reply:
No worries, as long as we can narrow it down - Bart and Johan may be able to prod at this if they have time first.